Essentials/UI/CooldownButton.cs

```diff
@@ -1,295 +1,226 @@
+            killButtonManager.renderer.enabled = true;
             Texture2D tex = GUIExtensions.CreateEmptyTexture();
-            byte[] buttonTexture = Reactor.Extensions.Extensions.ReadFully(resourceStream);
-            ImageConversion.LoadImage(tex, buttonTexture, false);
-            KillButtonManager.renderer.sprite = GUIExtensions.CreateSprite(tex);
-
-            PassiveButton button = KillButtonManager.GetComponent<PassiveButton>();
+            ImageConversion.LoadImage(tex, this.image, false);
+            killButtonManager.renderer.sprite = GUIExtensions.CreateSprite(tex);
+            PassiveButton button = killButtonManager.GetComponent<PassiveButton>();
             button.OnClick.RemoveAllListeners();
-            button.OnClick.AddListener(new Action(() =>
+            button.OnClick.AddListener((UnityEngine.Events.UnityAction)listener);
+            void listener()
             {
-                if (Timer < 0f && CanUse)
+                if (Timer < 0f && canUse)
                 {
-                    KillButtonManager.renderer.color = new Color(1f, 1f, 1f, 0.3f);
-
+                    killButtonManager.renderer.color = new Color(1f, 1f, 1f, 0.3f);
                     if (hasEffectDuration)
                     {
-                        IsEffectActive = true;
-
+                        isEffectActive = true;
                         Timer = EffectDuration;
-
-                        KillButtonManager.TimerText.Color = new Color(0, 255, 0);
+                        killButtonManager.TimerText.Color = new Color(0, 255, 0);
                     }
-
-                    onClick();
+                    OnClick();
                 }
-            }));
-        }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category, float effectDuration, Action onEffectEnd) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, true, effectDuration, onEffectEnd)
-        {
+            }
         }
-
-        /// <summary>
-        /// Call in HudManager.Start
-        /// </summary>
-        public CooldownButton(Action onClick, float cooldown, string imageEmbededResourcePath, Vector2 positionOffset, ButtonCategory category) : this(Assembly.GetCallingAssembly(), onClick, cooldown, imageEmbededResourcePath, positionOffset, category, false, 0F, null)
+        public bool CanUse()
         {
+            if (PlayerControl.LocalPlayer.Data == null) return false;
+            canUse = useTester();
+            return true;
         }
-
-        public bool IsUsable()
+        public static void HudUpdate()
         {
-            if (PlayerControl.LocalPlayer?.Data == null) return false;
-
-            switch (Category)
+            buttons.RemoveAll(item => item.killButtonManager == null);
```

Comment:
Nullifies the reason for `Start()`'s existence since the `CooldownButton` wrapper object becomes irrelevant once the wrapped `KillButtonManager` instance no longer exists. Yes this is an issue with the existing code too, but it is changing.